### ISSUE TITLE
mympirun fails with foss/2017a (OpenMPI 2.0x.x) 

### ISSUE THREAD

**Issue by Caylo** [hpcugent/vsc-mympirun#113]
```--------------------------------------------------------------------------
A hostfile was provided that contains at least one node not
present in the allocation:

  hostfile:  /user/home/gent/vsc413/vsc41353/.mympirun_7uzyhp/3152078.master15.delcatty.gent.vsc_20170322_112851/nodes
  node:      node2157

If you are operating in a resource-managed environment, then only
nodes that are in the allocation can be used in the hostfile. You
may find relative node syntax to be a useful alternative to
specifying absolute node names see the orte_hosts man page for
further information.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
ORTE was unable to reliably start one or more daemons.
This usually is caused by:

* not finding the required libraries and/or binaries on
  one or more nodes. Please check your PATH and LD_LIBRARY_PATH
  settings, or configure OMPI with --enable-orterun-prefix-by-default

* lack of authority to execute on one or more specified nodes.
  Please verify your allocation and authorities.

* the inability to write startup files into /tmp (--tmpdir/orte_tmpdir_base).
  Please check with your sys admin to determine the correct location to use.

*  compilation of the orted with dynamic libraries when static are required
  (e.g., on Cray). Please check your configure cmd line and consider using
  one of the contrib/platform definitions for your system type.

* an inability to create a connection back to mpirun due to a
  lack of common network interfaces and/or no route found between
  them. Please check network connectivity (including firewalls
  and network routing requirements).
--------------------------------------------------------------------------
```

this is possibly because the nodefile doesnt use fqdn (?).

**Comment by boegel** [comment]
The problem seems to be that OpenMPI 2.x uses the value returned by `hostname`, which in our systems returns something like `node1234.cluster.os`, while the `nodes` file contains `node1234.cluster.gent.vsc` (which is the FQDN on the IB network), see https://github.com/open-mpi/ompi/issues/2328#issuecomment-257762514

@stdweird Thoughts on this?

**Comment by boegel** [comment]
@Caylo I think it makes sense to make `mympirun` spit out a clear warning when it is being used with OpenMPI 2.0.x; the problem is fixed in OpenMPI 2.1.0 and is not there with OpenMPI 1.x.

**Comment by boegel** [comment]
fixed in #123
